src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -847,6 +847,40 @@ internal override TypeSymbol SetUnknownNullabilityForReferenceTypes()
             return result;
         }
 
+        internal override TypeSymbol SetPossiblyNullableReferenceTypeTypeParametersAsNullable()
+        {
+            if (!IsGenericType)
+            {
+                return this;
+            }
+
+            var allTypeArguments = ArrayBuilder<TypeSymbolWithAnnotations>.GetInstance();
+            GetAllTypeArgumentsNoUseSiteDiagnostics(allTypeArguments);
+
+            bool haveChanges = false;
+            for (int i = 0; i < allTypeArguments.Count; i++)
+            {
+                TypeSymbolWithAnnotations oldTypeArgument = allTypeArguments[i];
+                TypeSymbolWithAnnotations newTypeArgument = oldTypeArgument.SetPossiblyNullableReferenceTypeTypeParametersAsNullable();
+                if (!oldTypeArgument.IsSameAs(newTypeArgument))
+                {
+                    allTypeArguments[i] = newTypeArgument;
```

Comment:
>allTypeArguments[i] = newTypeArgument; [](start = 20, length = 38)

It feels like we should unconditionally store a value into `allTypeArguments[i]`, so that the array builder is completely initialized.